### PR TITLE
chore: fix 4.x-stable publish script

### PR DESCRIPTION
**Pull request by afc163**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [x] Workflow
- [ ] Other (about what?)

### 💡 Background and solution

修复 4.x-stable 分支发布版本时的一些问题。

- [x] conch 版本不应该更新 5.x 。
- [x] 把 5.x 里输入 code 前的系统通知挪过来。https://github.com/ant-design/ant-design/pull/41613
- [x] 通知更多钉钉群。
- [x] eslint 里忽略掉 master 分支上产生的一些文件和目录。 
- [x] 升级到 actions/checkout@v4 https://github.com/ant-design/ant-design/pull/44623

图片

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |      --     |
| 🇨🇳 Chinese |     --      |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at a1c9d92</samp>

Improved the version comparison logic in `scripts/post-script.js` to avoid unnecessary warnings. Updated the `dingding-token` parameter of the `release-helper` action in `.github/workflows/release-helper.yml` to support different types of DingTalk bots.

### <samp>🤖 Generated by Copilot at a1c9d92</samp>

* Update the `release-helper` action to use different DingTalk bot tokens for different user groups ([link](https://github.com/ant-design/ant-design/pull/44658/files?diff=unified&w=0#diff-9f6a6ebe440a447969c3bb7246d3c927c241861fdeb56b314a29b58dcf14d1e8L31-R31))
* Modify the `SAFE_DAYS_DIFF` function in `scripts/post-script.js` to filter out incompatible versions ([link](https://github.com/ant-design/ant-design/pull/44658/files?diff=unified&w=0#diff-0c429b9a01f20777bce85b2ad2472fe50f00b4ef8a404d56852f3d4dd15e905dR50))
* Remove an empty comment line from `scripts/post-script.js` for code style ([link](https://github.com/ant-design/ant-design/pull/44658/files?diff=unified&w=0#diff-0c429b9a01f20777bce85b2ad2472fe50f00b4ef8a404d56852f3d4dd15e905dL114-L116))

### CODE REVIEW

File: scripts/pre-publish-notice.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+const { Notification } = require('node-notifier');
+
+new Notification().notify({
+  title: '✅ 准备发布到 npm',
+  message: '测试用例执行完毕，马上就要输入 npm 校验码了！',
+  sound: true,
```
Comment (vagusX) [inline scripts/pre-publish-notice.js:6]: 这个厉害了

File: scripts/pre-publish-notice.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+const { Notification } = require('node-notifier');
+
+new Notification().notify({
+  title: '✅ 准备发布到 npm',
+  message: '测试用例执行完毕，马上就要输入 npm 校验码了！',
```
Comment (afc163) [inline scripts/pre-publish-notice.js:5]:
```suggestion
  message: '测试用例执行完毕，快回来输入 npm 校验码了！',
```

File: scripts/pre-publish-notice.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+const { Notification } = require('node-notifier');
+
+new Notification().notify({
+  title: '✅ 准备发布到 npm',
+  message: '测试用例执行完毕，快回来输入 npm 校验码了！',
```
Comment (yoyo837) [inline scripts/pre-publish-notice.js:5]:
```suggestion
  message: '测试用例执行完毕，快回来输入 npm 校验码！',
```

Comment (afc163) [reply scripts/pre-publish-notice.js:5]: 多个 `了` 貌似没啥毛病。

**Review by yoyo837 — APPROVED** [review]

**Pull request closed without merging**